music_assistant/providers/podcast-index/helpers.py

```diff
@@ -0,0 +1,223 @@
+    AudioFormat,
+    ItemMapping,
+    MediaItemImage,
+    Podcast,
+    PodcastEpisode,
+    ProviderMapping,
+    UniqueList,
+)
+
+from .constants import API_BASE_URL
+
+if TYPE_CHECKING:
+    from music_assistant.mass import MusicAssistant
+
+
+async def make_api_request(
+    mass: MusicAssistant,
+    api_key: str,
+    api_secret: str,
+    endpoint: str,
+    params: dict[str, Any] | None = None,
+) -> dict[str, Any]:
+    """
+    Make authenticated request to Podcast Index API.
+
+    Handles authentication using SHA1 hash of API key, secret, and timestamp.
+    Maps HTTP errors appropriately: 401 -> LoginFailed, others -> ProviderUnavailableError.
+    """
+    # Prepare authentication headers
+    auth_date = str(int(time.time()))
+    auth_string = api_key + api_secret + auth_date
+    auth_hash = hashlib.sha1(auth_string.encode()).hexdigest()
+
+    headers = {
+        "X-Auth-Key": api_key,
+        "X-Auth-Date": auth_date,
+        "Authorization": auth_hash,
+    }
+
+    url = f"{API_BASE_URL}/{endpoint}"
+
+    try:
+        async with mass.http_session.get(url, headers=headers, params=params or {}) as response:
+            response.raise_for_status()
+
+            try:
+                data: dict[str, Any] = await response.json()
+            except aiohttp.ContentTypeError as err:
+                raise InvalidDataError("Invalid JSON response from API") from err
+
+            if str(data.get("status")).lower() != "true":
+                raise InvalidDataError(data.get("description") or "API error")
+
+            return data
+
+    except aiohttp.ClientConnectorError as err:
+        raise ProviderUnavailableError(f"Failed to connect to Podcast Index API: {err}") from err
+    except aiohttp.ServerTimeoutError as err:
+        raise ProviderUnavailableError(f"Podcast Index API timeout: {err}") from err
+    except aiohttp.ClientResponseError as err:
+        if err.status == 401:
+            raise LoginFailed(f"Authentication failed: {err.status}") from err
+        raise ProviderUnavailableError(f"API request failed: {err.status}") from err
+
+
+def parse_podcast_from_feed(
+    feed_data: dict[str, Any], lookup_key: str, domain: str, instance_id: str
+) -> Podcast | None:
+    """Parse podcast from API feed data."""
+    feed_url = feed_data.get("url")
+    podcast_id = feed_data.get("id")
+
+    if not feed_url or not podcast_id:
+        return None
+
+    podcast = Podcast(
+        item_id=str(podcast_id),
+        name=feed_data.get("title", "Unknown Podcast"),
+        publisher=feed_data.get("author") or feed_data.get("ownerName", "Unknown"),
+        provider=lookup_key,
+        # uri=feed_data.get("url"),
+        provider_mappings={
+            ProviderMapping(
+                item_id=str(podcast_id),
+                provider_domain=domain,
+                provider_instance=instance_id,
+                url=None,  # Don't set URL to RSS feed - not directly playable
```

Comment:
this is fine, the url is only here for external referencing.

Reply:
This is not resolved. You can set the url here to the actual url...

```suggestion
                url= feed_url, 
```